fuels-core/src/code_gen/custom_types_gen.rs

```diff
@@ -227,32 +247,53 @@ mod tests {
     use std::str::FromStr;
 
     #[test]
-    #[should_panic(expected = "index out of bounds")]
-    fn test_extract_struct_name_from_abi_property_default() {
+    fn test_extract_custom_type_name_from_abi_property_bad_data() {
         let p: Property = Default::default();
-        let _ = extract_struct_name_from_abi_property(&p);
+        let result = extract_custom_type_name_from_abi_property(&p, "");
```

Comment:
Here, for instance, having the second arg be an enum will eliminate this type of bug completely!

Reply:
Yes!